tensorflow_addons/image/cutout_ops.py

```diff
@@ -30,11 +30,20 @@ def _get_image_wh(images, data_format):
 
 def _norm_params(images, mask_size, data_format):
     mask_size = tf.convert_to_tensor(mask_size)
-    if tf.rank(mask_size) == 0:
-        mask_size = tf.stack([mask_size, mask_size])
-    data_format = conv_utils.normalize_data_format(data_format)
-    image_height, image_width = _get_image_wh(images, data_format)
-    return mask_size, data_format, image_height, image_width
+    with tf.control_dependencies(
+        [
+            tf.assert_equal(
+                tf.reduce_any(mask_size % 2 != 0),
+                False,
+                "mask_size should be divisible by 2",
+            )
+        ]
+    ):
```

Comment:
control_dependencies is quite tf 1.x style, I recommend to activate the check only in eager mode. It will also be easier to debug for users.

Reply:
Your piece of code raised a very important question that I discussed in #1417 . We don't need to fix #1417 for merging your pull request of course.